python/tvm/auto_scheduler/task_scheduler.py

```diff
@@ -358,6 +358,12 @@ def tune(
         self.best_ct = self.ct
         self.best_score = self.cur_score
 
+        # put task without schedule on warm up to dead state
+        if self.strategy == "gradient":
```

Comment:
I feel this logic applies to all strategies instead of just gradient, so this condition may not be necessary. Could you help confirm?